impeller/entity/contents/filters/filter_contents.h

```diff
@@ -132,17 +132,23 @@ class FilterContents : public Contents {
   // |Contents|
   const FilterContents* AsFilter() const override;
 
-  /// @brief Determines the coverage of source pixels that will be needed
-  ///        to apply this filter under the given transform and produce
-  ///        results anywhere within the indicated coverage limit.
+  /// @brief  Determines the coverage of source pixels that will be needed
+  ///         to produce results for the indicated output limit. This is a
+  ///         reverse bounds calculation computing a source coverage from
+  ///         an intended output coverage.
   ///
-  ///        This is useful for subpass rendering scenarios where a filter
-  ///        will be applied to the output of the subpass and we need to
-  ///        determine how large of a render target to allocate in order
-  ///        to collect all pixels that might affect the supplied output
-  ///        coverage limit. While we might clip the rendering of the subpass,
-  ///        we want to avoid clipping out any pixels that contribute to
-  ///        the output limit via the filtering operation.
+  ///         The method computes a result such that if the filter is applied
+  ///         to a set of pixels filling the computed source coverage, it
+  ///         should produce an output that covers the entire indicated
+  ///         |output_limit|.
+  ///
+  ///         This is useful for subpass rendering scenarios where a filter
+  ///         will be applied to the output of the subpass and we need to
+  ///         determine how large of a render target to allocate in order
+  ///         to collect all pixels that might affect the supplied output
+  ///         coverage limit. While we might clip the rendering of the subpass,
+  ///         we want to avoid clipping out any pixels that contribute to
+  ///         the output limit via the filtering operation.
```

Comment:
This never defines what `effect_transform` is.  Looking at the implementation of some `GetFilterSourceCoverage` i know that only the basis is used.  That should be mentioned.

For `DirectionalGaussianBlurFilterContents::GetFilterSourceCoverage` shouldn't the result of `GetFilterSourceCoverage` be smaller than `output_limit` since the source will not have the blur radius?  It is actually returning a larger result.

The problem I ran into was this:
```c++
TEST(DirectionalGaussianBlurFilterContents, FilterSourceCoverage) {
  Scalar sigma_radius_1 = CalculateSigmaForBlurRadius(1.0);
  auto contents = std::make_unique<DirectionalGaussianBlurFilterContents>();
  contents->SetSigma(Sigma{sigma_radius_1});
  contents->SetDirection({1.0, 0.0});
  std::optional<Rect> coverage = contents->GetFilterSourceCoverage(
      /*effect_transform=*/Matrix::MakeScale({2.0, 2.0, 1.0}),
      /*output_limit=*/Rect::MakeLTRB(100, 100, 200, 200));
  ASSERT_EQ(coverage, ???);
}
```
Should the output_limit be `Rect::MakeLTRB(100 - 2, 100, 200 + 2, 200)` or `Rect::MakeLTRB(50 - 2, 50, 100 + 2, 100)` or `Rect::MakeLTRB(200 - 2, 200, 400 + 2, 400)`?  It's not clear how the `effect_transform` is related to the operation by reading the documentation.

Reply:
> This never defines what `effect_transform` is. Looking at the implementation of some `GetFilterSourceCoverage` i know that only the basis is used. That should be mentioned.

Use of Basis is specific to a filter computing its best guess, not to the general operation. All this method specifies is that the filters should do a reverse computation, not how they should do it.

For the record, I don't think the use of basis on those filters is correct and I should file an issue about it ([done](https://github.com/flutter/flutter/issues/137723)). I used basis because the existing methods in the class already used basis and I wanted to be consistent with the current state of their implementations.

> For `DirectionalGaussianBlurFilterContents::GetFilterSourceCoverage` shouldn't the result of `GetFilterSourceCoverage` be smaller than `output_limit` since the source will not have the blur radius? It is actually returning a larger result.

No, it has to be larger. Input pixels are taken from a radius and so we need more pixels from the source than the size of the dest.

> The problem I ran into was this:
> 
> ```c++
> TEST(DirectionalGaussianBlurFilterContents, FilterSourceCoverage) {
>   Scalar sigma_radius_1 = CalculateSigmaForBlurRadius(1.0);
>   auto contents = std::make_unique<DirectionalGaussianBlurFilterContents>();
>   contents->SetSigma(Sigma{sigma_radius_1});
>   contents->SetDirection({1.0, 0.0});
>   std::optional<Rect> coverage = contents->GetFilterSourceCoverage(
>       /*effect_transform=*/Matrix::MakeScale({2.0, 2.0, 1.0}),
>       /*output_limit=*/Rect::MakeLTRB(100, 100, 200, 200));
>   ASSERT_EQ(coverage, ???);
> }
> ```
> 
> Should the output_limit be `Rect::MakeLTRB(100 - 2, 100, 200 + 2, 200)` or `Rect::MakeLTRB(50 - 2, 50, 100 + 2, 100)` or `Rect::MakeLTRB(200 - 2, 200, 400 + 2, 400)`? It's not clear how the `effect_transform` is related to the operation by reading the documentation.

I mentioned that it was relative to the transform in the most recent cut, but I'll be more specific. The supplied "output_limit" and the result should both be in the transformed space. The sigma is in the untransformed space, but the input and output coverages are both in the transformed space. So, you need to adjust the sigma to its correct value in the transformed space and then all 3 are in the same space.

So, here you would need to scale the radius by 2, but then apply it directly to the 100,100 values and return them relative to 100,100 - so expand by 2 in both left and right. Your first answer looks right to me, other than assuming that sigma is the value you should expand to. There is a calculation in the code that maps sigma to the "fringe" of the gaussian blur. That "fringe" calculation will be in the untransformed space, so compute sigma -> fringe, and then map fringe through the transform.

But since both the input and the output rectangles are in the transformed space of the `effect_transform`, neither of those gets run through, or inversed through, the transform.